### PR TITLE
zlog 1.2.12 (new formula)

### PR DESCRIPTION
**Pull request by qinxin0720**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/zlog.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  url "https://github.com/HardySimpson/zlog/archive/1.2.12.tar.gz"
+  sha256 "9c6014a3f74d136c70255539beba11f30e1d3617d07ce7ea917b35f3e52bac20"
+
+  def install
+    system "make"
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+  
+  test do
+    system "#{bin}/zlog-chk-conf", "-h"
```
Comment (SMillerDev) [inline Formula/zlog.rb]: Could you make this test the function of the program? might be easiest to copy the instructions from https://github.com/HardySimpson/zlog/blob/master/README.md

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/zlog.rb (hunk 1 above)
Comment (qinxin0720) [reply Formula/zlog.rb]: I was made test function of this program yet.

File: Formula/zlog.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          return -2;
+        }
+
+        zlog_info(c, "hello, zlog!");
+        zlog_fini();
+
+        return 0;
+      }
+    EOS
+    system ENV.cc, "zlog.c", "-L#{lib}", "-lzlog", "-lpthread", "-o", "zlog"
```
Comment (DomT4) [inline Formula/zlog.rb]: It's a tiny tiny thing but for the sake of consistency can we stick to `"-o", "test"` and `system "./test"`, thanks. Just makes it easier to find all the instances at once when necessary/desirable.

File: Formula/zlog.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+class Zlog < Formula
+  desc "High-performance C logging library"
+  homepage "https://github.com/HardySimpson/zlog"
+  url "https://github.com/HardySimpson/zlog/archive/1.2.12.tar.gz"
+  sha256 "9c6014a3f74d136c70255539beba11f30e1d3617d07ce7ea917b35f3e52bac20"
+
+  def install
+    system "make"
```
Comment (DomT4) [inline Formula/zlog.rb]: Does this need a separate `make` step or will it compile & install fine with just:
```ruby
system "make", "PREFIX=#{prefix}", "install"
```
?

Comment (qinxin0720) [reply Formula/zlog.rb]: https://github.com/HardySimpson/zlog/blob/master/README.md#L23

Comment (qinxin0720) [reply Formula/zlog.rb]: It will compile with `system "make"` & install with `system "make", "PREFIX=#{prefix}", "install"`

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/11c546b470dd6e067315847eb8492b82fdee1b74. Thank you for your contribution to Homebrew @qinxin0720; we appreciate it! :smiley_cat:

**Pull request closed without merging**